### ISSUE TITLE
Feature Request: AGC for Microphone

### ISSUE THREAD

**Issue by cloudwindy** [OneOfEleven/uv-k5-firmware-custom#176]
Is it possible to automatically adjust mic gain based on sound level?

**Comment by cloudwindy** [comment]
Since I haven't come up with a practical solution, let's make this a discussion.

**Comment by OneOfEleven** [comment]
Yes it's easy to do. I'll add it at some point, busy though now with other things for a while.

**Comment by RoadRangerX** [comment]
I was thinking that an option to enable the Tx compression without the Rx expansion (Compandor On) would do this? I was just about to put that as a feature request when I saw this issue :-)

**Comment by OneOfEleven** [comment]
You can certainly enable the TX compander if you want, it will give more TX audio mic gain to those that want/need it.

**Comment by bricky149** [comment]
I believe this is possible without the compander. It's part of #218, AGC can control both MIC (Tx) and IF (Rx) inputs.

**Comment by RoadRangerX** [comment]
Does the Tx Compandor actually work? With the mic gain all the way up I still have low modulation and turning the Tx Compandor on doesn't seen to do anything? Any possibility of more gain in the firmware?
I'm using egzumer's build 0.14.1

**Comment by OneOfEleven** [comment]
Yes the compander works.

Have no idea why you have low TX audio. Just make sure you use WIDE mode with receiving radios that are set to wide mode.

**Comment by RoadRangerX** [comment]
This graph from audio recorded off my local repeater shows another operator first, me with Compandor on and max mic gain, and with Compandor off. Yes, the Tx Compandor does work - if I blow this up I can see it doing a bit of gain and limiting peaks.
[PoorModulation](https://github.com/OneOfEleven/uv-k5-firmware-custom/assets/24575272/7d9ef0c8-0bcf-464b-9d32-38ee0f530df9)

**Comment by OneOfEleven** [comment]
Is the latest firmware any better ?

**Comment by RoadRangerX** [comment]
> Is the latest firmware any better ?

I'm using egzumer's builds. I see he dropped release v0.15 yesterday, hopefully he brought in whatever change you made in your code?

Might take me a couple days - my 1TB SSD totally died this morning and I need to redo whatever I set up in the last 2 or three months which includes the UK-K5 software :-(

Thanks! Ron
